WalletWasabi.Fluent/Models/Wallets/Address.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Extensions;
+using WalletWasabi.Hwi;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.Models.Wallets;
+
+public class Address : ReactiveObject, IAddress
+{
+	public Address(Wallet wallet, HdPubKey hdPubKey)
+	{
+		Wallet = wallet;
+		HdPubKey = hdPubKey;
+		Network = wallet.Network;
+		HdFingerprint = Wallet.KeyManager.MasterFingerprint;
+		BitcoinAddress = HdPubKey.GetAddress(wallet.Network);
+	}
+
+	public Wallet Wallet { get; }
+	public HdPubKey HdPubKey { get; }
+	public Network Network { get; }
+	public HDFingerprint? HdFingerprint { get; }
+	public BitcoinAddress BitcoinAddress { get; }
+	public SmartLabel Label => HdPubKey.Label;
+	public PubKey PubKey => HdPubKey.PubKey;
+	public KeyPath FullKeyPath => HdPubKey.FullKeyPath;
+	public string Text => BitcoinAddress.ToString();
+	public IEnumerable<string> Labels => Label;
+
+	private bool IsUnused => !Label.IsEmpty && !HdPubKey.IsInternal && HdPubKey.KeyState == KeyState.Clean;
+
+	public bool IsUsed => !IsUnused;
```

Comment:
This doesn't look okay to me because here `IsUsed` is true for `Used` keys and `Locked` keys (Unused).

Reply:
You're right. `IsUsed` should be `!IsUnused && !IsLocked` or something. Will fix that.

Reply:
My suggestion is to fix the logic in a different PR and import it later here.